Orange/widgets/data/owtable.py

```diff
@@ -46,9 +46,14 @@
                                                  ANNOTATED_DATA_SIGNAL_NAME)
 from Orange.widgets.utils.itemmodels import TableModel
 from Orange.widgets.utils.state_summary import format_summary_details
+from Orange.widgets.utils.concurrent import ConcurrentMixin
+from orangewidget.utils.cache import LRUCache
+
 
+Viewport = namedtuple("Viewport", ("top", "left", "bottom", "right"))
 
-class RichTableModel(TableModel):
+
+class RichTableModel(TableModel, ConcurrentMixin):
```

Comment:
This class now inherits ConcurrentMixin. I think you should instead make a new class that handles concurrent stuff and then you encapsulate it into the class as an attribute, which you call. Reading this code is now slightly harder because (in many cases optional) concurrency handling functions (which you even can not rename but to how the Mixin works) are mixed with others.

Reply:
I suggest also including the cache within that new class.

Reply:
Cool, this looks much more self-contained. Thanks!